hack/make-rules/test-e2e-node.sh

```diff
@@ -48,11 +48,16 @@ timeout_arg=""
 system_spec_name=${SYSTEM_SPEC_NAME:-}
 extra_envs=${EXTRA_ENVS:-}
 runtime_config=${RUNTIME_CONFIG:-}
-ssh_user=${SSH_USER:-"${USER}"}
 ssh_key=${SSH_KEY:-}
 ssh_options=${SSH_OPTIONS:-}
 kubelet_config_file=${KUBELET_CONFIG_FILE:-"test/e2e_node/jenkins/default-kubelet-config.yaml"}
 
+if [[ ${KUBE_SSH_USER} != "" ]]; then
```

Comment:
`hack/lib/init.sh` has `set -o nounset` and `set -o errexit`, so this line will exit if `KUBE_SSH_USER` is not set.  You probably want to guard against this case, which you can do using `:-`:
```suggestion
if [[ "${KUBE_SSH_USER:-}" != "" ]]; then
```
or
```suggestion
if [[ -n "${KUBE_SSH_USER:-}" ]]; then
```
This whole if block could be replaced with the following:
```shell
ssh_user=${KUBE_SSH_USER:-$USER}
```
This would still error out in the unlikely event that `USER` were not set, which I assume is what you want.